Site/ASAB Maestro/Files/influxdb/influx-init.sh

```diff
@@ -0,0 +1,36 @@
+#!/bin/sh
+
+# Wait for InfluxDB to start
+until curl -s http://{{INFLUXDB_HOSTNAME}}:8086/health | grep -q '"status": "pass"'; do
+  echo "Waiting for InfluxDB to start..."
+  sleep 1
+done
+
+# Get the bucket ID
+BUCKET_ID=$(curl -s -X GET http://{{INFLUXDB_HOSTNAME}}:8086/api/v2/buckets -H 'Authorization: Token "{{INFLUXDB_TOKEN}}"' -H "Accept: application/json" | jq -r '.buckets[] | select(.name=="{{DOCKER_INFLUXDB_INIT_BUCKET}}") | .id')
+# Get the org ID
+ORG_ID=$(curl -s -X GET http://{{INFLUXDB_HOSTNAME}}:8086/api/v2/orgs -H 'Authorization: Token "{{INFLUXDB_TOKEN}}"' -H "Accept: application/json" | jq -r '.orgs[] | select(.name=="{{DOCKER_INFLUXDB_INIT_ORG}}") | .id')
+
+# Get the user ID
+USER_ID=$(curl -s -X GET http://{{INFLUXDB_HOSTNAME}}:8086/api/v2/users -H 'Authorization: Token "{{INFLUXDB_TOKEN}}"' -H "Accept: application/json" | jq -r '.users[] | select(.name=="{{DOCKER_INFLUXDB_INIT_USERNAME}}") | .id')
+
+# Create the non-admin token using the InfluxDB API v2
+curl -X POST http://{{INFLUXDB_HOSTNAME}}:8086/api/v2/authorizations \
```

Comment:
we need to get the token itself and send it to vault.

Reply:
even if you manage to send the token into vault, I don't know how to effectively distribute it. 
but I can probably somehow solve this in the influxdb tech.